src/partisan_config.erl

```diff
@@ -37,25 +37,31 @@ set(Key, Value) ->
 peer_config() ->
     Port = case partisan_config:get(peer_port, undefined) of
         undefined ->
-            %% Generate a random peer port.
-            rand_compat:seed(erlang:phash2([node()]),
-                             erlang:monotonic_time(),
-                             erlang:unique_integer()),
-            RandomPeerPort = rand_compat:uniform(1000) + 10000,
+            case application:get_env(partisan, peer_port) of
```

Comment:
The issue was that it relied on the DCOS os env. But instead of just changing it in that section I left that as it is and instead made it check if when `partisan_config` has `peer_port` of `undefined` and `peer_port` is defined in the env, use that value. So if the user doesn't want to use a static peer port they simply don't set one.